bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/YamlModelRepositoryImpl.java

```diff
@@ -153,13 +155,19 @@ public void deactivate() {
         watchService.unregisterListener(this);
     }
 
+    // Method for unit tests
+    protected void ignoreFolderRestriction() {
+        folderRestrictionIgnored = true;
+    }
+
     // The method is "synchronized" to avoid concurrent files processing
     @Override
     @SuppressWarnings({ "rawtypes", "unchecked" })
     public synchronized void processWatchEvent(Kind kind, Path path) {
         Path fullPath = watchPath.resolve(path);
         String modelName = path.toString();
-        if (path.startsWith("automation") || !modelName.endsWith(".yaml")) {
+        if (!modelName.endsWith(".yaml") || (!folderRestrictionIgnored
+                && !RESTRICTED_SUB_FOLDERS.stream().anyMatch(sf -> path.startsWith(sf)))) {
```

Comment:
- I think we should not log when the event is fired on files outside the Restricted folders. It floods the log when we only want to pay attention to the trace logs for the directories we _are_ actually watching. I worked on this before seeing your PR and in my version I called it `WATCHED_FOLDERS` instead. It more directly conveys the meaning but this is subjective.



- It would be more efficient if RESTRICTED_SUB_FOLDERS is a set of Path instead of string, therefore the `startsWith` operation doesn't have to keep converting String to Path on each event invocation.

Reply:
Changed